lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -783,6 +786,155 @@ void LifetimeChecker::diagnoseInitError(const DIMemoryUse &Use,
     diagnose(Module, TheMemory.getLoc(), diag::variable_defined_here, isLet);
 }
 
+static void injectHopToExecutorBefore(SILLocation loc,
+                                      SILBasicBlock::iterator &insertPt,
+                                      SILValue actor, bool needsBorrow) {
+  SILBuilder B(insertPt);
+
+  LLVM_DEBUG(llvm::dbgs() << "hop-injector: inserting hop_to_executor before ");
+  LLVM_DEBUG(insertPt->dump());
+
+  if (needsBorrow) {
+    auto *borrow = B.createBeginBorrow(loc, actor);
+    assert(borrow->getNumResults() == 1);
+    actor = borrow->getResult(0);
+  }
+
+  B.createHopToExecutor(loc, actor, /*mandatory=*/false);
+
+  if (needsBorrow)
+    B.createEndBorrow(loc, actor);
+}
+
+void LifetimeChecker::injectActorHops() {
+  auto ctor = TheMemory.isActorInitSelf();
+
+  // Must be `self` within an actor's designated initializer.
+  if (!ctor)
+    return;
+
+  // If the initializer has restricted use of `self`, then no hops are needed.
+  if (isRestrictedActorInitSelf())
+    return;
+
+  // Even if there are no stored properties to initialize, we still need a hop.
+  // We insert this directly after the mark_uninitialized instruction, so
+  // that it happens as early as `self` is available.`
+  if (TheMemory.getNumElements() == 0) {
+    auto selfDef = TheMemory.getUninitializedValue();
+    auto insertPt = selfDef->getNextInstruction()->getIterator();
+    return injectHopToExecutorBefore(ctor, insertPt, selfDef,
+                                     /*needsBorrow=*/true);
+  }
+
+  // A collection of pairs: an initializing block and its in-availability set.
+  using Vec = llvm::SmallVector<std::pair<SILBasicBlock *, AvailabilitySet>, 8>;
```

Comment:
Is this needed? Do we get more than 80 columns here? It really doesn't add anything. Also can you extract this into a class level data structure that we can clear. That way if we go big, we can just reuse memory.

Reply:
I just like type aliases for long types, and hadn't settled on a good name for it yet.

Do you really want me to move it into the LifetimeChecker for optimization purposes? What if I just made it a `static` decl in the function? Would that avoid the allocation only until the first time `injectHops` is called?

Reply:
I have instead simply eliminated the data structure 😄